lib/lobby.js

```diff
@@ -0,0 +1,55 @@
+const { sleep } = require("./helpers");
+
+class Lobby {
+  constructor(waitingPlayers = [], games = new Set()) {
+    this.waitingPlayers = waitingPlayers;
+    this.games = games;
+  }
+  addGame(game) {
+    this.games.add(game);
+  }
+  addAsWaitingPlayer(player) {
+    if (!this.waitingPlayers.includes(player)) {
+      player.log("Adding you to the lobby");
+      this.waitingPlayers.push(player);
+    }
+  }
+  getGameFromPlayer(player) {
```

Comment:
(optional) since game is not stored on the player, you're not getting it *from* the player:
```suggestion
  getGameForPlayer(player) {
```